frost-core/src/benches.rs

```diff
@@ -0,0 +1,201 @@
+//! Ciphersuite-generic benchmark functions.
+
+use std::collections::HashMap;
+
+use criterion::{BenchmarkId, Criterion, Throughput};
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{batch, frost, Ciphersuite, Signature, SigningKey, VerifyingKey};
+
+struct Item<C: Ciphersuite> {
+    vk: VerifyingKey<C>,
+    sig: Signature<C>,
+}
+
+fn sigs_with_distinct_keys<C: Ciphersuite, R: RngCore + CryptoRng + Clone>(
+    rng: &mut R,
+) -> impl Iterator<Item = Item<C>> {
+    let mut rng = rng.clone();
+    std::iter::repeat_with(move || {
+        let msg = b"Bench";
+        let sk = SigningKey::new(&mut rng);
+        let vk = VerifyingKey::from(&sk);
+        let sig = sk.sign(&mut rng, &msg[..]);
+        Item { vk, sig }
+    })
+}
+
+/// Benchmark batched signature verification with the specified ciphersuite.
+pub fn bench_batch_verify<C: Ciphersuite, R: RngCore + CryptoRng + Clone>(
+    c: &mut Criterion,
+    name: &str,
+    rng: &mut R,
+) {
+    let mut group = c.benchmark_group(format!("Batch Verification {}", name));
+    for &n in [8usize, 16, 24, 32, 40, 48, 56, 64].iter() {
+        group.throughput(Throughput::Elements(n as u64));
+
+        let sigs = sigs_with_distinct_keys::<C, R>(rng)
+            .take(n)
+            .collect::<Vec<_>>();
+
+        group.bench_with_input(
+            BenchmarkId::new("Unbatched verification", n),
+            &sigs,
+            |b, sigs| {
+                b.iter(|| {
+                    for item in sigs.iter() {
+                        let msg = b"Bench";
+
+                        let Item { vk, sig } = item;
+                        {
+                            let _ = vk.verify(msg, sig);
+                        }
```

Comment:
###### Optional simplification
```suggestion
                        let _ = vk.verify(msg, sig);
```